src/typ.rs

```diff
@@ -277,16 +277,16 @@ pub enum ValueConstructorVariant {
     Record {
         name: String,
         field_map: Option<FieldMap>,
-        arity: usize,
     },
 }
 
 impl ValueConstructorVariant {
     fn to_module_value_constructor(&self) -> ModuleValueConstructor {
         match self {
-            ValueConstructorVariant::Record { name, .. } => {
-                ModuleValueConstructor::Record { name: name.clone() }
-            }
+            ValueConstructorVariant::Record { name, field_map } => ModuleValueConstructor::Record {
+                name: name.clone(),
+                arity: field_map.clone().map_or(0, |fm| fm.arity),
```

Comment:
Could we map without cloning? No need to clone the entire field map if we only the arity from it.

Reply:
Excellent suggestion

Reply:
Done